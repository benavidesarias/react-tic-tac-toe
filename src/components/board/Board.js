import React from 'react';
import Square from '../square/Square';
import calculateWinner from '../../util/util';

class Board extends React.Component {

    renderSquare(i, className) {

        return (
            <Square
                key={i}
                className={className}
                value={this.props.squares[i]}
                onClick={() => this.props.onClick(i)}
            />
        );
    }

    render() {

        const rows = [];
        let squareIndex = 0;

        const winner = calculateWinner(this.props.squares);

        // 3x3 Squares 
        for (let row = 0; row < 3; row++) {
            const children = [];
            for (let col = 0; col < 3; col++) {

                if (winner.winner && (squareIndex === winner.row[0] || squareIndex === winner.row[1] || squareIndex === winner.row[2]))
                    children.push(this.renderSquare(squareIndex, "winner-square"));
                else
                    children.push(this.renderSquare(squareIndex, "square"));

                squareIndex++;
            }
            rows.push(<div className="board-row" key={row}>{children}</div>);
        }

        return (
            <div>
                {rows}
            </div>
        );
    }
}

export default Board;