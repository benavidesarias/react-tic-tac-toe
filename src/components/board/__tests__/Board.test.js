import React from 'react';
import ReactDOM from 'react-dom';
import Board from '../Board';

import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

it("render", () =>{
    const squares= Array(9).fill(null);

    const div = document.createElement("div");
    ReactDOM.render(<Board squares={squares}/>,div);
});

it("snapshot some movement", () =>{
    const squares= Array(9).fill(null);
    squares[0] = 'X';
    squares[1] = 'O';
    squares[2] = 'X';

    const tree = renderer.create(<Board squares={squares}/>).toJSON();
    expect(tree).toMatchSnapshot(); 
});