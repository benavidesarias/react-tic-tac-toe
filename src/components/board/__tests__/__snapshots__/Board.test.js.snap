// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot some movement 1`] = `
<div>
  <div
    className="board-row"
  >
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
  </div>
</div>
`;
